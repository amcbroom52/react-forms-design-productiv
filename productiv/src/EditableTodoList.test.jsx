import { describe, it, expect } from "vitest";

import React from "react";
import { fireEvent, render } from "@testing-library/react";
import EditableTodoList from "./EditableTodoList";
import EditableTodo from "./EditableTodo";


const TODOS = [
    {
    id: '1',
    title: "title1",
    description: "description1",
    priority: 1
    },
    {
    id: '2',
    title: "title2",
    description: "description2",
    priority: 2
    },
    {
    id: '3',
    title: "title3",
    description: "description3",
    priority: 3
    }
];

describe("EditableTodoList component", function () {
    it("renders without crashing", function () {
        render(<EditableTodoList todos={TODOS} />);
    });

    it("matches snapshot", function () {
        const { asFragment } = render(<EditableTodoList todos={TODOS} />);
        expect(asFragment()).toMatchSnapshot();
    })

    it("contains correct Todo values", function() {
        const {container} = render(<EditableTodoList todos={TODOS} />);
        expect(container).toContainHTML("description1");
        expect(container).toContainHTML("description2");
        expect(container).toContainHTML("description3");
    });

    it("contains correct Todo values", function() {
        const {container} = render(<EditableTodoList todos={TODOS} />);
        let editButtons = container.querySelectorAll(".EditableTodo-toggle")
        expect(editButtons.length).toEqual(3);

        fireEvent.click(editButtons[0]);
        editButtons = container.querySelectorAll(".EditableTodo-toggle")
        expect(editButtons.length).toEqual(2);
    });
})
