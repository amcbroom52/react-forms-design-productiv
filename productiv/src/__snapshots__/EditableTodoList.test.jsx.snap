// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EditableTodoList component > matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
          id="1"
        >
          <div>
            <b>
              title1
            </b>
             
            <small>
              Priority: 1
            </small>
          </div>
          <div>
            <small>
              description1
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
          id="2"
        >
          <div>
            <b>
              title2
            </b>
             
            <small>
              Priority: 2
            </small>
          </div>
          <div>
            <small>
              description2
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
          id="3"
        >
          <div>
            <b>
              title3
            </b>
             
            <small>
              Priority: 3
            </small>
          </div>
          <div>
            <small>
              description3
            </small>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Todo component > matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
          id="1"
        >
          <div>
            <b>
              title1
            </b>
             
            <small>
              Priority: 1
            </small>
          </div>
          <div>
            <small>
              description1
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
          id="2"
        >
          <div>
            <b>
              title2
            </b>
             
            <small>
              Priority: 2
            </small>
          </div>
          <div>
            <small>
              description2
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
          id="3"
        >
          <div>
            <b>
              title3
            </b>
             
            <small>
              Priority: 3
            </small>
          </div>
          <div>
            <small>
              description3
            </small>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
